lib/collection/src/shard/transfer/shard_transfer.rs

```diff
@@ -263,13 +264,59 @@ pub async fn promote_proxy_to_remote_shard(
     }
 }
 
+/// Activate peer for replica.
+///
+/// Returns true if the replica was enabled, false if it was already handled
+pub async fn activate_peer_for_replica(
+    shard_holder: Arc<LockedShardHolder>,
+    shard_id: ShardId,
+    replica_peer: PeerId,
+) -> CollectionResult<bool> {
+    let mut shard_holder_guard = shard_holder.write().await;
+    let shard = shard_holder_guard.get_mut_shard(&shard_id);
+    match shard {
+        Some(Shard::ReplicaSet(replica_set)) => {
+            // already activated
+            if replica_set.replica_state.peers.get(&replica_peer) == Some(&true) {
+                Ok(false)
+            } else {
+                replica_set.set_active(&replica_peer, true)?;
+                Ok(true)
+            }
```

Comment:
We should mark it active only through consensus operation as this should be globally synchronized

Reply:
This happens at the end of the transfer when `finish_shard_transfer` is triggered via consensus. It should be good as well.